src/dbnode/persist/fs/seek_manager.go

```diff
@@ -498,6 +504,68 @@ func (m *seekerManager) updateOpenLeaseHotSwapSeekers(
 	return wg, updateOpenLeaseResult, nil
 }
 
+func (m *seekerManager) RelinquishShard(shard uint32) error {
+	byTime := m.seekersByTime(shard)
+
+	// Indicate this shard is closed so it can't be borrowed and no seekers
+	// can be opened from now on.
+	byTime.Lock()
+	byTime.closed = true
+	byTime.Unlock()
+
+	for _, seekers := range byTime.seekers {
```

Comment:
I think you need to be holding the lock the entire time you do this so maybe don't relinquish it above on line 514